### ISSUE TITLE
collectChecks is now upstream

### ISSUE THREAD

**Issue by dhess** [hackworthltd/haskell-hacknix#103]
Here's ours:

https://github.com/hackworthltd/haskell-hacknix/blob/8a83b364dcb853215138a09966906561bf0c7a79/nix/overlays/haskell.nix#L180

Here's upstream's:

https://github.com/input-output-hk/haskell.nix/blob/c88f9eccc975b21ae1e6a6b8057a712b91e374f2/lib/default.nix#L183

**Comment by dhess** [comment]
Note that it's really a drop-in replacement, because we like to filter package sets by package name (or prefix, e.g., "hhp*"), the upstream `collectChecks` only passes the package to the filter (not `name: pkg:`), and some packages have missing `identifer.name` attributes; e.g.:

```
error: --- ThrownError ---------------------------------------------------------------------------------------------------------------------------------------------------------- nix-build
The option `packages.Win32.package.identifier.name' is used but not defined.
```

However, we can probably use the `haskell-nix.haskellLib.isLocal` filter to replace name/prefix filtering.

**Comment by dhess** [comment]
Closed by 776fadbef32cea3cdf0baad3cca02885560d7850
